scheduler/src/Control/Scheduler.hs

```diff
@@ -260,9 +290,11 @@ scheduleWorkId_ = _scheduleWorkId
 --
 -- @since 1.4.1
 replicateWork :: Applicative m => Int -> Scheduler m a -> m a -> m ()
-replicateWork !n scheduler f
-  | n <= 0 = pure ()
-  | otherwise = scheduleWork scheduler f *> replicateWork (pred n) scheduler f
+replicateWork !n scheduler f = go n
+  where
+    go k
+      | k <= 0 = pure ()
+      | otherwise = scheduleWork scheduler f *> go (k - 1)
```

Comment:
Is `k - 1` better than `pred`? And the `k` should probably be banged.

Reply:
Here is the implementation of `pred`:
```haskell
    pred x
       | x == minBound  = errorWithoutStackTrace "Prelude.Enum.pred{Int}: tried to take `pred' of minBound"
       | otherwise      = x - 1
```
So, yeah, it is better for two reasons:
* We avoid a partial function
* We avoid an unnecessary branch

Bang won't give any improvement really, since we force `k` at each iteration of the loop with the guard, but I'll add it anyways.

Reply:
A nice piece of Haskell trivia. Thanks!